webextension/scripts/settings.js

```diff
@@ -88,6 +99,68 @@ function selectall () {
   }
 }
 
+function validatePrivateMode (event) {
+  let checkboxes = $('[name="private-include"]')
+  let checkedCount = checkboxes.filter((_index, item) => item.checked === true).length
+  if (checkedCount > 0) {
+    $('#private-mode').prop('checked', false)
+  } else {
+    $('#private-mode').prop('checked', true)
+  }
+
+  // Checking previous state when the only one item is checked and remove 'selected-prior' class from everywhere
+  if (checkedCount === 0 && $('.selected-prior').length === 1) {
+    $('.selected-prior').removeClass('selected-prior')
+  }
+
+  // If the event.taget.checked is true,then check if it the checkbox is only item, then remove 'selected-prior' from all
+  // And add class 'selected-prior' to the event.taget, if it is NOT there
+  // Else If the event.taget.checked is false, check if the check-box is NOT the only item, then remove class 'selected-prior'
+  if (event.target.checked === true) {
```

Comment:
The way some of this is being handled doesn't seem right to me, that is showing and hiding things like `#borrow_books` or `#wayback-count-label` should be written once in the code where settings are set or cleared, and the only code that should be in this function is to set or clear those settings. This may involve reorganizing code so that each setting as a function to clear and set each, but that organization would be preferable than to have code copied in this function.